cishouseholds/weights/weights.py

```diff
@@ -414,43 +437,69 @@ def validate_design_weights(
     )
     df = assign_distinct_count_in_group(
-        df, "ANTIBODY_DISTINCT_DESIGN_WEIGHT_BY_GROUP", [swab_weight_column], antibody_group_by_columns
+        df, "ANTIBODY_DISTINCT_DESIGN_WEIGHT_BY_GROUP", [antibody_weight_column], antibody_group_by_columns
     )
     swab_design_weights_inconsistent_within_group = (
         False if df.filter((F.col("SWAB_DISTINCT_DESIGN_WEIGHT_BY_GROUP") > 1)).count() == 0 else True
     )
     antibody_design_weights_inconsistent_within_group = (
         False if df.filter((F.col("ANTIBODY_DISTINCT_DESIGN_WEIGHT_BY_GROUP") > 1)).count() == 0 else True
     )
-
     df.drop(
         "SWAB_DESIGN_WEIGHT_SUM_CHECK_FAILED",
         "ANTIBODY_DESIGN_WEIGHT_SUM_CHECK_FAILED",
         "SWAB_DISTINCT_DESIGN_WEIGHT_BY_GROUP",
         "ANTIBODY_DISTINCT_DESIGN_WEIGHT_BY_GROUP",
     )
     error_string = ""
+
+    if not swab_weight_column_type:
+        error_string += "\n- Swab design weights are not DecimalType."
+    if not antibody_weight_column_type:
+        error_string += "\n- Antibody design weights are not DecimalType."
+
     if not antibody_design_weights_sum_to_population:
-        error_string += "Antibody design weights do not sum to country population totals.\n"
+        error_string += "\n- Antibody design weights do not sum to country population totals."
     if not swab_design_weights_sum_to_population:
-        error_string += "Swab design weights do not sum to cis area population totals.\n"
-    if positive_design_weights > 0:
-        error_string += f"{positive_design_weights} records have negative design weights.\n"
+        error_string += "\n- Swab design weights do not sum to cis area population totals."
+    if negative_design_weights > 0:
+        error_string += f"\n- {negative_design_weights} records have negative design weights."
+        check_negative_design_weights = True
+    else:
+        check_negative_design_weights = False
+
     if null_design_weights > 0:
-        error_string += f"There are {null_design_weights} records with null swab or antibody design weights.\n"
+        error_string += f"\n- There are {null_design_weights} records with null swab or antibody design weights."
+        check_null_design_weights = True
+    else:
+        check_null_design_weights = False
+
     if swab_design_weights_inconsistent_within_group:
-        error_string += "Swab design weights are not consistent within CIS area population groups.\n"
+        error_string += "\n- Swab design weights are not consistent within CIS area population groups."
     if antibody_design_weights_inconsistent_within_group:
-        error_string += "Antibody design weights are not consistent within country population groups.\n"
+        error_string += "\n- Antibody design weights are not consistent within country population groups."
+
+    if error_string:
+        error_string += "\n"
     if error_string:
         raise DesignWeightError(error_string)
+    return (
```

Comment:
We don't want to have the code return something just for the sake of testing. Please see the docs for `pytest.raises`, where you can use regex to check the error message for specific text.